app/controllers/concerns/search_result_helper.rb

```diff
@@ -0,0 +1,50 @@
+module SearchResultHelper
+  class << self
+    # A method that uses the CardComponentSerializer to create search results
+    #
+    # @param [Array<Object>] results an array of results that is used to create search results
+    def create_search_results(results)
+      results.entries.map do |entry|
+        data = {
+          'heading-text':   entry.title,
+          url:              entry.links.first,
+          'list-hint':      list_hint_data(entry),
+          'paragraph-text': entry.content
+        }
+
+        ComponentSerializer::CardComponentSerializer.new('card__search-result', data).to_h
+      end
+    end
+
+    private
+
+    def list_hint_data(entry)
```

Comment:
You're throwing around an awful lot of random hashes - can these data structures be formalised into classes, services, etc?

Looking at these methods its very difficult to follow what the end result is. Documentation would help, but not necessarily fix the problem